subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testsuites/TestSuitesDependenciesIntegrationTest.groovy

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing.testsuites
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+
+
+class TestSuitesDependenciesIntegrationTest extends AbstractIntegrationSpec {
+
+  def 'suites do not share dependencies by default'() {
+      given:
+      buildFile << """
+        plugins {
+          id 'java'
+        }
+
+        ${mavenCentralRepository()}
+
+        testing {
+            suites {
+                test {
+                    dependencies {
+                        implementation 'org.apache.commons:commons-lang3:3.11'
+                    }
+                }
+                integTest(JvmTestSuite) {
+                    useJUnit()
+                }
+            }
+        }
+
+        tasks.named('check') {
+            dependsOn testing.suites.integTest
+        }
+        """
+
+      file('src/test/java/example/UnitTest.java') << '''
+            package example;
+
+            import org.apache.commons.lang3.StringUtils;
+            import org.junit.Assert;
+            import org.junit.Test;
+
+            public class UnitTest {
+                @Test
+                public void unitTest() {
+                    Assert.assertTrue(StringUtils.isEmpty(""));
+                }
+            }
+        '''
+
+      file('src/integTest/java/it/IntegrationTest.java') << '''
+            package it;
+
+            import org.apache.commons.lang3.StringUtils; // compilation fails here; commons-lang3 is not automatically "inherited" by integTests
+            import org.junit.Assert;
+            import org.junit.Test;
+
+            public class IntegrationTest {
+                @Test
+                public void integrationTest() {
+                    Assert.assertTrue(StringUtils.isEmpty("")); // compilation also fails here; commons-lang3 is not automatically "inherited" by integTests
+                }
+            }
+        '''
+
+      when:
+      fails 'check'
+
+      then:
+      failureCauseContains('Compilation failed; see the compiler error output for details.')
```

Comment:
This is OK, but it could be easy to accidentally make `IntegrationTest` fail for other reasons than a missing dependency or even fail compilation for `UnitTest`. 

I would either narrow the task you run to just the integTest compilation task and check for a particular error in the output or change this around to resolve the integTest's compileClasspath and runtimeClasspath and show that it doesn't contain commons-lang3.

Reply:
I agree, "failing" tests can give false signal for any other number of reasons. Do you think it's better to run `integTestCompile` explicitly, or implement the `checkConfiguration` custom task pattern from `TestSuitesIntegrationTest`? I'm OK with either pattern.

Reply:
I would probably do the `checkConfiguration` way because it also lets you remove the "noise" around having test code at all.  We can just check the dependencies are as expected.

Reply:
Fixed; much, much cleaner now.